istioctl/cmd/proxystatus_v2.go

```diff
@@ -0,0 +1,151 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"text/tabwriter"
+	"time"
+
+	xdsapi "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	corev2 "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/spf13/cobra"
+
+	"istio.io/istio/istioctl/pkg/clioptions"
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/adsc"
+)
+
+const (
+	connectionsTypeURL = "istio.io/connections"
+)
+
+func newProxyStatusCommand() *cobra.Command {
+	var opts clioptions.CentralControlPlaneOptions
+
+	statusCmd := &cobra.Command{
+		Use:   "proxy-status [<pod-name[.namespace]>]",
+		Short: "Retrieves the synchronization status of each Envoy in the mesh",
+		Long: `
+Retrieves last sent and last acknowledged xDS sync from Pilot to each Envoy in the mesh
+
+`,
+		Example: `# Retrieve sync status for all Envoys in a mesh
+	istioctl proxy-status
+
+# Retrieve sync diff for a single Envoy and Pilot
+	istioctl proxy-status productpage-v1-679f4bcbbb-6j4j6.default
+`,
+		Aliases: []string{"ps"},
+		RunE: func(c *cobra.Command, args []string) error {
+			if opts.Xds == "" {
+				return fmt.Errorf("--endpoint default not implemented; MUST be supplied; try localhost:15012")
+			}
+			adscConn, err := adsc.Dial(opts.Xds, "", &adsc.Config{
+				IP: "1.2.3.4",
```

Comment:
shouldn't this be the endpoint passed in?

Reply:
I don't think we use the IP any more - we may want to use a special constant to indicate it's from istioctl.

The labels are important however - if we want Istioctl to display the config to be generated for a workload by istiod. This will need a flag.